projects/packages/search/src/initializers/class-jetpack-initializer.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Search initializer for the Jetpack plugin.
+ *
+ * @package    @automattic/jetpack-search
+ */
+
+namespace Automattic\Jetpack\Search;
+
+/**
+ * Initializer for the main Jetpack plugin.
+ * Instantiate this initializer to enable Jetpack Search functionality.
+ */
+class Jetpack_Initializer extends Initializer {
+	/**
+	 * Initializes either the Classic Search or the Instant Search experience.
+	 */
+	public static function initialize() {
+		// Check whether Jetpack Search should be initialized in the first place.
+		if ( ! self::is_connected() || ! self::is_search_supported() ) {
+			/**
+			 * Fires when the Jetpack Search fails and would fallback to MySQL.
+			 *
+			 * @since Jetpack 7.9.0
+			 * @param string $reason Reason for Search fallback.
+			 * @param mixed  $data   Data associated with the request, such as attempted search parameters.
+			 */
+			do_action( 'jetpack_search_abort', 'inactive', null );
+			return;
+		}
+
+		$blog_id = \Jetpack::get_option( 'id' );
+		if ( ! $blog_id ) {
+			do_action( 'jetpack_search_abort', 'no_blog_id', null );
+			return;
+		}
+
+		// TODO: Port the search widget to package for milestone 2.
+		require_once JETPACK__PLUGIN_DIR . 'modules/widgets/search.php';
+
+		/**
+		 * Location of built instant search assets, does not include trailing slash.
+		 */
+		define( 'JETPACK_SEARCH_PLUGIN_DIRECTORY', dirname( dirname( __DIR__ ) ) );
+
+		if ( Options::is_instant_enabled() ) {
+			// Enable the instant search experience.
+			Instant_Search::initialize( $blog_id );
+
+			// Register instant search configurables as WordPress settings.
+			new Settings();
+
+			// Instantiate "Customberg", the live search configuration interface.
+			Customberg::instance();
+
+			// Enable configuring instant search within the Customizer.
+			if ( class_exists( 'WP_Customize_Manager' ) ) {
+				// TODO: Port this class to the package.
+				require_once JETPACK__PLUGIN_DIR . 'modules/search/class-jetpack-search-customize.php';
+				new \Jetpack_Search_Customize();
+			}
+		} else {
+			// Enable the classic search experience.
+			Classic_Search::initialize( $blog_id );
+		}
+	}
+
+	/**
+	 * Check if site has been connected.
+	 */
+	public static function is_connected() {
+		return \Jetpack::is_connection_ready();
```

Comment:
We probably want to use the `Automattic\Jetpack\Connection\Manager` rather than `Jetpack` which is only available in the plugin.

Reply:
Also I can still see several places referencing `Jetpack`, which might need to be replaced/copied from Jetpack plugin too.

Reply:
It shouldn't matter for this specific class, since this initializer is actually intended to be used only within the Jetpack plugin context.